lib/openstack_query/queries/query_wrapper.py

```diff
@@ -15,18 +15,13 @@ class QueryWrapper(QueryMethods, QueryBase):
     QueryWrapper is a base class for all Query<Resource> subclasses
     """
 
-    def __init__(self, runner_cls):
-        prop_handler = self._get_prop_handler()
-
-        self.runner = runner_cls(
-            marker_prop_func=lambda obj: prop_handler.get_prop(obj, self.marker_enum)
-        )
+    def __init__(self):
         self._query_results = []
 
-        self.output = QueryOutput(prop_handler)
-        self.parser = QueryParser(prop_handler)
+        self.output = QueryOutput(self.prop_enum_cls)
+        self.parser = QueryParser(self.prop_enum_cls)
         self.builder = QueryBuilder(
```

Comment:
Maybe a future PR, but these should move into QueryBase and there should be a factory pattern

Something like

```
class GenericQueryFactory:
  @static_method
  def build_composition(prop_cls):
      output = QueryOutput(prop_cls)
      parser = QueryParser(prop_cls)
      builder = ...

     return QueryBaseDeps(output, parser, builder)  # new dataclass

```
      
```
class ServerQuery(QueryBase, QueryMethods):
    # This allows us to inject a mock class instead
    def __init__(base_factory = GenericQueryFactory):
        super().__init__(base_factory.build_composition(ServerProperties))
```

(And it merges query_wrapper and query_base together)